doc/stdenv/platform-notes.chapter.md

```diff
@@ -59,79 +57,59 @@ Some common issues when packaging software for Darwin:
 
-  ```nix
-  stdenv.mkDerivation {
-    name = "libfoo-1.2.3";
-    # ...
-    env.PACKAGE_SPECIFIC_SDK_VAR = apple-sdk_10_15.sdkroot;
-    # or
-    buildInputs = [ apple-sdk_10_15 ];
-    postPatch = ''
-      export PACKAGE_SPECIFIC_SDK_VAR=$SDKROOT
-    '';
-  }
-  ```
+When determining whether to use a non-default SDK, consider the following:
 
-  The following is a list of Xcode versions, the SDK version in nixpkgs, and the attribute to use to add it. Generally, only the last SDK release for a major version is packaged (each _x_ in 10._x_ until 10.15 is considered a major version).
+- Try building your derivation with the default SDK. If it works, you’re done.
+- If the package specifies a specific version, use that. See below for how to map Xcode version to SDK version.
+- If the package’s documentation indicates it supports optional features on newer SDKs, consider using the SDK that enables those features. If you’re not sure, use the default SDK.
 
-  | Xcode version      | SDK version                                       | nixpkgs attribute |
-  |--------------------|---------------------------------------------------|-------------------|
-  | Varies by platform | 10.12.2 (x86_64-darwin)<br/>11.3 (aarch64-darwin) | `apple-sdk`       |
-  | 8.0–8.3.3          | 10.12.2                                           | `apple-sdk_10_12` |
-  | 9.0–9.4.1          | 10.13.2                                           | `apple-sdk_10_13` |
-  | 10.0–10.3          | 10.14.6                                           | `apple-sdk_10_14` |
-  | 11.0–11.7          | 10.15.6                                           | `apple-sdk_10_15` |
-  | 12.0–12.5.1        | 11.3                                              | `apple-sdk_11`    |
-  | 13.0–13.4.1        | 12.3                                              | `apple-sdk_12`    |
-  | 14.0–14.3.1        | 13.3                                              | `apple-sdk_13`    |
-  | 15.0–15.4          | 14.4                                              | `apple-sdk_14`    |
-  | 16.0               | 15.0                                              | `apple-sdk_15`    |
+Note: It is possible to have multiple, different SDK versions in your inputs. When that happens, the one with the highest version is always used.
 
-  To use a non-default SDK, add it to your build inputs.
+##### Propagating an SDK (advanced, compilers-only)
 
-  ```nix
-  stdenv.mkDerivation {
-    name = "libfoo-1.2.3";
-    # ...
-    buildInputs = [ apple-sdk_15 ]; # Uses the 15.0 SDK instead of the default SDK for the platform.
-  }
-  ```
+The SDK is a package, and it can be propagated. However, most packages should *not* do this. The exception is compilers. When you propagate an SDK, it becomes part of your derivation’s public API, and changing the SDK or removing it can be a breaking change. That is why propagating it is only recommended for compilers.
```

Comment:
This should go below stuff like minimum versions and be combined with the section on propagating a minimum version, I think. Most users will never have to think about this. We should also clarify that even for language toolchains you should only do this if actually necessary.